dex/order/order.go

```diff
@@ -298,108 +285,129 @@ type MarketOrder struct {
 	Filled uint64
 }
 
-// ID computes the order ID.
-func (o *MarketOrder) ID() OrderID {
-	if o.id != nil {
-		return *o.id
-	}
-	id := calcOrderID(o)
-	o.id = &id
-	return id
-}
+// T is an alias for Trade. Embedding with the alias allows us to define a
+// method on the interface called Trade that returns the *Trade.
+type T = Trade
 
-// UID computes the order ID, returning the string representation.
-func (o *MarketOrder) UID() string {
-	if o.uid != "" {
-		return o.uid
-	}
-	uid := o.ID().String()
-	o.uid = uid
-	return uid
+// Trade returns a pointer to the orders embedded Trade.
+func (t *Trade) Trade() *Trade {
+	return t
```

Comment:
Yah, `func (p *Prefix) Prefix() *Prefix { return p }` is peculiar, but it works and a simpler and clearer solution isn't jumping out at me.